src/utilities/FieldPlaneAveraging.H

```diff
@@ -0,0 +1,81 @@
+//
+//  FieldPlaneAveraging.H
+//  amr-wind
+//
+
+#ifndef FieldPlaneAveraging_H
+#define FieldPlaneAveraging_H
+
+
+#include <AMReX_AmrCore.H>
+#include "DirectionSelector.H"
+#include "Field.H"
+
+class incflo;
+
+class FieldPlaneAveraging
+{
+public:
+    FieldPlaneAveraging(amr_wind::Field &field, int axis_in = 2);
+
+    ~FieldPlaneAveraging(){};
+
+    void operator()();
+
+    amrex::Real eval_line_average(amrex::Real x, int comp) const; /** evaluate line average at specific location for any average component */
+    amrex::Real eval_line_average(int cell, int comp) const; /** evaluate line average at specific for any average component */
+
+    /** output a single text file with averages only */
+//    void
+//    plot_line_average_text(std::string filename, int step, amrex::Real time);
+//    /** output a single text file with averages and fluctuations */
+//    void plot_line_text(std::string filename, int step, amrex::Real time);
+//    /** output a single binary file with averages and fluctuations */
+//    void plot_line_binary(std::string filename, int step, amrex::Real time);
+//
+//    /** output a line plot file based on the plot type */
+//    void plot_line(int step, amrex::Real time, int plot_type = 0);
+
+    /** change precision of text file output */
+    void set_precision(int p) {precision = p;};
+
+    amrex::Real get_dx(){return dx;};
+    amrex::Real get_xlo(){return xlo;};
```

Comment:
For all methods 

- Use `const` qualifiers 
- Prefer `dx()` instead of `get_*`